### PR TITLE
V2? 

### PR DESCRIPTION
**Pull request by IT-MikeS**
Some changes, one big one that I think warrants a version bump from 1.x.x to 2.x.x.

- I change the application over to using a module that actually downloads and converts on the machine itself instead of relying on the web-service as it could one day no longer exist.

- Implemented the dialog from electron to give better save location control to the user on any electron supported platform.

- Fixed odd rendering on the percentage on the progress bar where the % symbol would fall outside the bar.

- Cleaned up unused code and dependencies after the switch to the module. (no more axios, or need of ipcRenderer)

What still needs to be done:

- Comment code better to reflect new changes.

- Test on platforms other than Windows

- Better error handling and testing the errors that may occur.

Hope this help!

### CODE REVIEW

File: app/containers/app.container.js [hunk 1]
```diff
@@ -1,102 +1,107 @@
-import React, { Component } from 'react';
-import Axios from 'axios';
+import React, {Component} from 'react';
 import LinkInput from '../components/LinkInput';
 import ProgressBar from '../components/ProgressBar';
 
-const { ipcRenderer } = window.require('electron')
+const ytdl = window.require('ytdl-core');
+const fs = window.require('fs');
+import * as path from "path";
```
Comment (leerob) [inline app/containers/app.container.js]: Nit: Prefer using single quotes here

File: app/containers/app.container.js [hunk 2]
```diff
@@ -1,102 +1,107 @@
...
+  getVideoAndConvert(urlLink, userProvidedPath, title){
+    this.setState({progressMessage: 'Downloading and converting...'});
+    return new Promise((resolve, reject) => {
+      let fullPath = path.join(userProvidedPath, `${title}.mp3`);
+      let videoObject = ytdl(urlLink, {filter: 'audioonly'});
+
+      videoObject
+        .on('progress', (chunkLength, downloaded, total) => {
+          if(!this.rateLimitTrigged) {
+            let newval = Math.floor((downloaded / total) * 100).toString();
```
Comment (leerob) [inline app/containers/app.container.js]: `newVal` might be better here.

File: app/containers/app.container.js [hunk 3]
```diff
@@ -1,102 +1,107 @@
...
+
+      videoObject
+        .pipe(fs.createWriteStream(fullPath))
+        .on('finish', () => {
+          resolve();
+        });
+    });
+  }
+
+  async ConvertVideoToMp3(urlLink, userProvidedPath) {
```
Comment (leerob) [inline app/containers/app.container.js]: I think this function should start with a lower case C

**Comment by IT-MikeS** [comment]
I agree with your code change suggestions and I'll reflect those asap. As for the location to downloads, I like the idea that you came up with in having it as a setting that the user can select but will default to the download folder for the appropriate platform the app is being run on.

**Comment by leerob** [comment]
Awesome, yeah I think that’s a good approach. I’ll be able to test the new functionality tomorrow and let you know how it works.

**Comment by IT-MikeS** [comment]
Awesome, I'll start working on changes and the user settings stuff

**Comment by IT-MikeS** [comment]
Closed will submit new PR with master branch

**Comment by leerob** [comment]
Sounds good, thanks!

**Pull request closed without merging**